### ISSUE TITLE
Slow on foldmethod=syntax

### ISSUE THREAD

**Issue by phonglk** [sbdchd/neoformat#340]
**TL;DR;** install https://github.com/Konfekt/FastFold

Not an actual problem of neoformat.
When you are using formatted with foldmethod=syntax, it is drastically slow.
e.g: For an 800 lines ts file, prettier took 500ms to produce output but using neoformat would take 4s to response
After debugging, I found it is because of setline fn (https://github.com/sbdchd/neoformat/blob/master/autoload/neoformat.vim#L125) and particularly slow when foldmethod=indent
The issue also mentioned here https://github.com/dense-analysis/ale/issues/1829
After installing install https://github.com/Konfekt/FastFold, the issue is gone.

This is not a bug ticket, but informational ticket. May include in the doc.
